lib/OpenQA/Utils.pm

```diff
@@ -310,7 +311,26 @@ sub bugurl {
 sub bugref_to_href {
     my ($text) = @_;
 
-    $text =~ s{((bnc|bsc|boo|poo)#(\d+))}{<a href="@{[bugurl($2)]}$3">$1</a>}gi;
+    $text =~ s{((bnc|bsc|boo|poo)#(\d+)( |$))}{<a href="@{[bugurl($2)]}$3">$1</a>}gim;
+
+    return $text;
+}
+
+sub bugabbr {
+    my ($bugurl) = @_;
+    my %bugurls = (
+        'https://bugzilla.novell.com/show_bug.cgi?id='   => 'bnc',
+        'https://bugzilla.suse.com/show_bug.cgi?id='     => 'bsc',
+        'https://bugzilla.opensuse.org/show_bug.cgi?id=' => 'boo',
+        'https://progress.opensuse.org/issues/'          => 'poo',
+    );
```

Comment:
Instead of duplicating the hash from lines 303-306 and turning it around, can you define the hash just once and use it inversely here?